pkg/geo/geomfn/subdivide.go

```diff
@@ -0,0 +1,364 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"math"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/errors"
+	"github.com/twpayne/go-geom"
+)
+
+// Subdivide returns a geometry divided into parts, where each part contains no more than the number of vertices provided.
+func Subdivide(g geo.Geometry, maxVertices int) (geo.Geometry, error) {
+	if g.Empty() {
+		return g, nil
+	}
+	const minMaxVertices = 5
+	if maxVertices < minMaxVertices {
+		return geo.Geometry{}, errors.Newf("max_vertices number cannot be less than %v", minMaxVertices)
+	}
+
+	gt, err := g.AsGeomT()
+	if err != nil {
+		return geo.Geometry{}, errors.Newf("could not transform input geometry into geom.T: %v", err)
+	}
+	col := geom.NewGeometryCollection()
+	dim, err := dimensionFromGeomT(gt)
+	if err != nil {
+		return geo.Geometry{}, errors.Newf("could not calculate geometry dimension: %v", err)
+	}
+	const startDepth = 0
+	err = subdivideRecursive(gt, maxVertices, startDepth, col, dim)
+	if err != nil {
+		return geo.Geometry{}, err
+	}
+	col.SetSRID(gt.SRID())
```

Comment:
nit: this i think can be the same line as NewGeometryCollection:
```
col := geom.NewGeometryCollection().SetSRID(srid)
```

Reply:
Fixed